crates/forge/bin/cmd/verify/etherscan/mod.rs

```diff
@@ -327,16 +342,39 @@ impl EtherscanVerificationProvider {
         let project = config.project()?;
 
         let contract_path = self.contract_path(args, &project)?;
-        let compiler_version = self.compiler_version(args, &config, &project)?;
-        let (source, contract_name, code_format) =
-            self.source_provider(args).source(args, &project, &contract_path, &compiler_version)?;
+        let mut compiler_version = self.compiler_version(args, &config, &project)?;
+        let zk_compiler_version = self.zk_compiler_version(args, &config, &project)?;
+
+        let source_provider = self.source_provider(args);
+        let (source, contract_name, code_format) = if let Some(zk) = &zk_compiler_version {
+            source_provider.zk_source(args, &project, &contract_path, &zk.zksolc)
+        } else {
+            source_provider.source(args, &project, &contract_path, &compiler_version)
+        }?;
+
+        let zk_args = match zk_compiler_version {
+            None => vec![],
+            Some(zk) => {
+                if let Some(solc) = zk.solc {
+                    compiler_version = Version::new(solc.major, solc.minor, solc.patch);
+                }
+
+                let compilermode = if zk.is_zksync_solc { "zksync" } else { "solc" }.to_string();
```

Comment:
We can use `compiler_mode` here for consistency, as `compilermode` is merely a serialized representation of it.